plugins/woocommerce/src/Internal/Admin/ProductForm/ComponentTrait.php

```diff
@@ -0,0 +1,59 @@
+<?php
+/**
+ * Product Form Traits
+ */
+
+namespace Automattic\WooCommerce\Internal\Admin\ProductForm;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * TaskTraits class.
+ */
+trait ComponentTrait {
+	/**
+	 * Component ID.
+	 *
+	 * @var string
+	 */
+	protected $id;
+
+	/**
+	 * Plugin ID.
+	 *
+	 * @var string
+	 */
+	protected $plugin_id;
```

Comment:
Cool!  Just out of curiosity, do we have plans as for how this might be used?  I assume the "slug/id" will be self-reported by the plugin adding it, right?

Reply:
> do we have plans as for how this might be used?

No not yet, I assume we might want to have an alternate function to register a plugin that may include an `icon` and a full name. Or we did want to force this to the actual `plugin` slug and check with the plugin list.
We haven't fully thought this through yet, but maybe matching this with the `plugin` slug for now would be a smart plan.

> I assume the "slug/id" will be self-reported by the plugin adding it, right?

Yeah, as there is no way to tell otherwise what plugin registered a field. So technically they could reference `woocommerce` as well, unless we prevent that from being allowed.